### ISSUE TITLE
[Meds / Improvements] Investigate client-side filter/sort

### ISSUE THREAD

**Issue by ckeif-iat** [department-of-veterans-affairs/va.gov-team#121179]
## Context 
This came out of this [Slack discussion](https://ironarchtechnologyhq.slack.com/archives/C096TAVFQPM/p1759241175103069?thread_ts=1758915811.453359&cid=C096TAVFQPM) around the Filter and Sort controls of the medications list view.

## Background
While discussing the accessibility changes being made around these two controls, it was brought up that it would be ideal to remove the loading spinner. This cannot be done today because a network request is made to handle the sort/filter operation on the backend. This task is to investigate if it's possible to move these operations to the frontend.

## Description 
Investigate how the sorting and filtering works today, and if it would be possible to load all of the data in the frontend then perform the filtering/sorting operations client-side.

## Resources
- [Original Slack discussion](https://ironarchtechnologyhq.slack.com/archives/C096TAVFQPM/p1759241175103069?thread_ts=1758915811.453359&cid=C096TAVFQPM)

## Tasks
- [ ] Investigate and document findings
- [ ] Create dev / implementation ticket(s)

## Acceptance Criteria
- [ ] Findings are documented
- [ ] Created dev / implementation ticket(s) if necessary
- [ ] Peer review
